server/src/main/java/org/elasticsearch/common/blobstore/fs/FsBlobStore.java

```diff
@@ -84,7 +89,7 @@ public void close() {
 
     private synchronized Path buildAndCreate(BlobPath path) throws IOException {
         Path f = buildPath(path);
-        if (!readOnly) {
+        if (readOnly == false) {
```

Comment:
Maybe we should also just throw here if `f` doesn't exist as well? I think that would make things a little easier to debug down the line instead of just quietly creating the unusable `BlobContainer` in `#blobcontainer` above and then throwing on all the operations?

Reply:
Hmm. I know what you mean, but today it seems to be ok to make a blob container on a nonexistent path and to treat it as if it is empty. See e.g. `FsBlobStoreTests#testReadOnly` that asserts that this is the desired behaviour. I can't immediately see any problems with changing this, but I think it needs more analysis to be sure.

Reply:
Ah right nevermind then :) I guess not throwing here is more aligned with blob stores and the URL repository anyway.
I was only thinking about making the case of e.g. trying to load an index that doesn't exist (i.e. not even its folder exists) a little easier to debug but you're right it's not worth the behaviour change :)